app/src/main/java/org/apache/fineract/ui/online/customers/customertasks/CustomerTasksBottomSheetFragment.java

```diff
@@ -125,6 +127,18 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
                 break;
         }
 
+        dialog.setOnKeyListener((dialog1, keyCode, event) -> {
+            if (keyCode == KeyEvent.KEYCODE_BACK) {
+                if (event.getAction() == KeyEvent.ACTION_DOWN) {
+                    if (isTaskFormVisible()) {
+                        hideTaskForm();
+                        return true;
+                    }
+                }
+            }
+            return false;
+        });
+
```

Comment:
@EGOR-IND isn't it possible to reduce code by using `onBackPressed()`? Also we can remove `llTaskList.getVisibility() == View.GONE` as both are contradictory anyone of them will do the same.